app/src/main/java/com/example/drawingApp/ColorPickerUtil.kt

```diff
@@ -0,0 +1,202 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.util.Log
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtil {
+    const val MAX_HUE = 360f
+    const val SAT_VAL_FACTOR = 100
+
+    data class ColorPack(
+        val hsv: Hsv,
+        val previousColor: Int
+    )
+
+    fun rgbToHsv(red: Int, green: Int, blue: Int): FloatArray {
```

Comment:
Returning an HSV here would be nicer I think

Reply:
implemented